### PR TITLE
Annevk/noncharacter

### PR DESCRIPTION
**Pull request by annevk**
***
[Preview](https://infra.spec.whatwg.org/branch-snapshots/annevk/noncharacter/) | [Diff](https://s3.amazonaws.com/pr-preview/whatwg/infra/c484a4f...60d532a.html)

### CODE REVIEW

**Comment by annevk** [comment]
It's not entirely clear to me what the best place for noncharacter is. Before we continue with this PR I'd first like to work on corresponding PRs for URL and HTML to make sure this actually helps.

**Comment by annevk** [comment]
Commit message note: this fixes #108.

**Comment by annevk** [comment]
HTML could maybe use a definition of controls that excludes ASCII whitespace.

**Comment by annevk** [comment]
I think it helps in terms of readability in both, but someone else should give it a look over as well.

File: infra.bs [hunk 1]
```diff
@@ -325,6 +325,12 @@ inclusive.
 
 <p>A <dfn export>scalar value</dfn> is a <a>code point</a> that is not a <a>surrogate</a>.
 
+<p>A <dfn export>noncharacter</dfn> is a <a>code point</a> that is in the range U+FDD0 to U+FDEF,
+inclusive, or U+FFFE, U+FFFF, U+1FFFE, U+1FFFF, U+2FFFE, U+2FFFF, U+3FFFE, U+3FFFF, U+4FFFE,
+U+4FFFF, U+5FFFE, U+5FFFF, U+6FFFE, U+6FFFF, U+7FFFE, U+7FFFF, U+8FFFE, U+8FFFF, U+9FFFE, U+9FFFF,
+U+AFFFE, U+AFFFF, U+BFFFE, U+BFFFF, U+CFFFE, U+CFFFF, U+DFFFE, U+DFFFF, U+EFFFE, U+EFFFF, U+FFFFE,
+U+FFFFF, U+10FFFE, or U+10FFFF.
```
Comment (domenic) [inline infra.bs]: Can we explain why these characters are chosen? Apparently they are "permanently undefined" per Unicode?

Still not sure about whether we should be defining explicit ranges or referencing Unicode.

Comment (annevk) [inline infra.bs]: Opened #117 to sort this out, but I don't think that needs to block this, as we don't explain "surrogate" or "scalar value" in much detail either.

**Pull request merged**